KD63799/deadlinedrive
# Add files for testing

This pull request introduces the following changes:

Installation of Jest:

Jest has been added to the project's dependencies to enable unit testing. This includes necessary configurations for Jest in package.json and the creation of a jest.config.js file.
Added the jest-environment-jsdom package to provide a browser-like environment for testing React components.
Basic Unit Tests:

Basic unit tests have been written for selected components and services to verify their functionality. These tests cover scenarios such as rendering, state changes, and service methods.
Installed Jest and jest-environment-jsdom packages.
Added Jest configuration in jest.config.js.
Created /tests/unit/components and /tests/unit/services directories.
Added example unit test files and basic tests.
How to Test:

Run npm install to install the new dependencies.
Execute npm test to run the Jest test suite and verify that all tests pass successfully.
Additional Information:
This setup provides a foundation for unit testing in the project. Future pull requests can build upon this by adding more comprehensive tests as new features are developed.
